File: src/application/use_cases/commands/category_with_tasks.rs
use crate::domain::entities::category::Category;
use crate::domain::entities::task::Task;

pub struct CategoryWithTasks {
    pub category: Category,
    pub tasks: Vec<Task>,
}

pub struct CategoryAndTasks {
    pub category_with_tasks: Vec<CategoryWithTasks>,
    pub orphan_tasks: Vec<Task>,
}

File: app/lib/domain/entities/category_with_tasks.dart
import 'package:equatable/equatable.dart';
import 'package:focus_flow_app/domain/entities/category.dart';
import 'package:focus_flow_app/domain/entities/task.dart';

class CategoryWithTasks extends Equatable {
  final Category category;
  final List<Task> tasks;

  const CategoryWithTasks({required this.category, required this.tasks});

  @override
  List<Object?> get props => [category, tasks];
}
